src/input/keyboard.rs

```diff
@@ -107,128 +103,90 @@
+            pressed_physical_keys: HashSet::with_capacity(256),
+            pressed_logical_keys: HashSet::with_capacity(256),
             last_pressed: None,
             current_pressed: None,
-            previously_pressed_keys_set: HashSet::with_capacity(256),
-            previously_pressed_scancodes_set: HashSet::with_capacity(256),
+            previously_pressed_physical_keys: HashSet::with_capacity(256),
+            previously_pressed_logical_keys: HashSet::with_capacity(256),
         }
     }
 
     /// Checks if a key is currently pressed down.
-    pub fn is_key_pressed(&self, key: KeyCode) -> bool {
-        self.pressed_keys_set.contains(&key)
+    pub fn is_logical_key_pressed(&self, key: &Key) -> bool {
+        self.pressed_logical_keys.contains(key)
     }
 
     /// Checks if a key has been pressed down this frame.
-    pub fn is_key_just_pressed(&self, key: KeyCode) -> bool {
-        self.pressed_keys_set.contains(&key) && !self.previously_pressed_keys_set.contains(&key)
+    pub fn is_logical_key_just_pressed(&self, key: &Key) -> bool {
+        self.pressed_logical_keys.contains(key)
+            && !self.previously_pressed_logical_keys.contains(key)
     }
 
     /// Checks if a key has been released this frame.
-    pub fn is_key_just_released(&self, key: KeyCode) -> bool {
-        !self.pressed_keys_set.contains(&key) && self.previously_pressed_keys_set.contains(&key)
+    pub fn is_logical_key_just_released(&self, key: &Key) -> bool {
+        !self.pressed_logical_keys.contains(key)
+            && self.previously_pressed_logical_keys.contains(key)
     }
 
     /// Checks if a key with the corresponding scan code is currently pressed down.
-    pub fn is_scancode_pressed(&self, code: ScanCode) -> bool {
-        self.pressed_scancodes_set.contains(&code)
+    pub fn is_physical_key_pressed(&self, physical_key: &PhysicalKey) -> bool {
+        self.pressed_physical_keys.contains(physical_key)
     }
 
     /// Checks if a key with the corresponding scan code has been pressed down this frame.
-    pub fn is_scancode_just_pressed(&self, code: ScanCode) -> bool {
-        self.pressed_scancodes_set.contains(&code)
-            && !self.previously_pressed_scancodes_set.contains(&code)
+    pub fn is_physical_key_just_pressed(&self, physical_key: &PhysicalKey) -> bool {
+        self.pressed_physical_keys.contains(physical_key)
+            && !self.previously_pressed_physical_keys.contains(physical_key)
     }
 
     /// Checks if a key with the corresponding scan code has been released this frame.
-    pub fn is_scancode_just_released(&self, code: ScanCode) -> bool {
-        !self.pressed_scancodes_set.contains(&code)
-            && self.previously_pressed_scancodes_set.contains(&code)
+    pub fn is_physical_key_just_released(&self, physical_key: &PhysicalKey) -> bool {
+        !self.pressed_physical_keys.contains(physical_key)
+            && self.previously_pressed_physical_keys.contains(physical_key)
```

Comment:
`PhysicalKey` is `Copy`, it should maybe be passed by value. Is guess that you kept a reference for consistency with logical key functions ?